crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1238 @@
+    ///     TranspilerError: If an operation class is passed in for ``instruction`` and no name
+    ///         is specified or ``properties`` is set.
+    #[pyo3(signature = (instruction, name, is_class, properties=None))]
+    fn add_instruction(
+        &mut self,
+        _py: Python<'_>,
+        instruction: GateRep,
+        name: String,
+        is_class: bool,
+        properties: Option<PropsMap>,
+    ) -> PyResult<()> {
+        if self.gate_map.contains_key(&name) {
+            return Err(PyAttributeError::new_err(format!(
+                "Instruction {:?} is already in the target",
+                name
+            )));
+        }
+        let mut qargs_val: PropsMap = PropsMap::new();
+        if is_class {
+            qargs_val = IndexMap::from_iter([(None, None)].into_iter());
+            self.variable_class_operations.insert(name.clone());
+        } else if let Some(properties) = properties {
+            let inst_num_qubits = instruction.num_qubits.unwrap_or_default();
+            if properties.contains_key(&None) {
+                self.global_operations
+                    .entry(inst_num_qubits)
+                    .and_modify(|e| {
+                        e.insert(name.clone());
+                    })
+                    .or_insert(HashSet::from_iter([name.clone()]));
+            }
+            for qarg in properties.keys() {
+                let mut qarg_obj = None;
+                if let Some(qarg) = qarg {
+                    if qarg.len() != inst_num_qubits {
+                        return Err(TranspilerError::new_err(format!(
+                            "The number of qubits for {name} does not match\
+                             the number of qubits in the properties dictionary: {:?}",
+                            qarg
+                        )));
+                    }
+                    self.num_qubits =
+                        Some(self.num_qubits.unwrap_or_default().max(
+                            qarg.iter().fold(
+                                0,
+                                |acc, x| {
+                                    if acc > x.index() {
+                                        acc
+                                    } else {
+                                        x.index()
+                                    }
+                                },
+                            ) + 1,
+                        ));
+                    qarg_obj = Some(qarg.clone())
+                }
+                qargs_val.insert(qarg_obj.to_owned(), properties[qarg].clone());
+                self.qarg_gate_map
+                    .entry(qarg_obj)
+                    .and_modify(|e| {
+                        if let Some(e) = e {
+                            e.insert(name.clone());
+                        }
+                    })
+                    .or_insert(Some(HashSet::from([name.clone()])));
+            }
+        }
+        // Modify logic once gates are in rust.
```

Comment:
Just so we have a common keyword to search for.
```suggestion
        // TODO: Modify logic once gates are in rust.
```